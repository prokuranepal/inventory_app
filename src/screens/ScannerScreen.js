import React, { useState, useEffect } from 'react';
import { Text, View, StyleSheet, Button,Platform } from 'react-native';
import { BarCodeScanner } from 'expo-barcode-scanner';
import HeaderButton from '../components/Component/HeaderButton';
import { HeaderButtons, Item } from 'react-navigation-header-buttons';
import Colors from '../constants/Colors'
// import * as itemsActions from '../store/actions/items';
import { useDispatch } from 'react-redux';

const ScannerScreen = props => {
    const dispatch = useDispatch();
    const [items, setitems] = useState();
    const [hasPermission, setHasPermission] = useState(null);
    const [scanned, setScanned] = useState(false);

    useEffect(() => {
        (async () => {
            const { status } = await BarCodeScanner.requestPermissionsAsync();
            setHasPermission(status === 'granted');
        })();
    }, []);

    const handleBarCodeScanned = ({ type, data }) => {
        setScanned(true);
        alert(`Bar code has data ${data} has been scanned!`);
        // setitems(items.push('Pantaprazole', '5', 'BGuyz', '30',
        //     'Paracetamol (acetaminophen) is a pain reliever and a fever reducer. The exact mechanism of action of is not known. Paracetamol is used to treat many conditions such as headache, muscle aches, arthritis, backache, toothaches, colds, and fevers.',
        //     'General'))

        setitems(data);
        // console.log("item", data)
        // dispatch(
        //     itemsActions.addItems(
        //         data,
        //         'BGuys',
        //         5,
        //         'Paracetamol (acetaminophen) is a pain reliever and a fever reducer',
        //         30,
        //         'General'

        //     )
        // );
        // alert(`Items has been added to inventory`);
        props.navigation.navigate
            ({
                routeName: 'ReceivedItemList',
                params: {
                    data: data
                }
            })

    };

    if (hasPermission === null) {
        return <Text>Requesting for camera permission</Text>;
    }
    if (hasPermission === false) {
        return <Text>No access to camera</Text>;
    }

    return (
        <View
            style={{
                flex: 1,
                flexDirection: 'column',
                justifyContent: 'flex-end',
            }}>
            <BarCodeScanner
                onBarCodeScanned={scanned ? undefined : handleBarCodeScanned}
                data-test="barComp"
                style={StyleSheet.absoluteFillObject}
            />

            {scanned && <Button title={'Tap to Scan Again'} onPress={() => setScanned(false)} />}
        </View>
    );
}

ScannerScreen.navigationOptions = navData => {
    return {
        headerTitle: "Scan to Add item in inventory",
        headerStyle: {
            backgroundColor: Platform.OS === 'android' ? Colors.primaryColor : ''
        },
        headerTitleStyle: {
            fontFamily: 'open-sans'
        },
        // headerLeft: () => (
        //     <HeaderButtons HeaderButtonComponent={HeaderButton} >
        //         <Item title="Menu" iconName="ios-menu" onPress={() => {
        //             navData.navigation.toggleDrawer()
        //         }} />
        //     </HeaderButtons>
        // )
    }
}

const styles = StyleSheet.create({
    screen: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    }
    ,

})

export default ScannerScreen;
