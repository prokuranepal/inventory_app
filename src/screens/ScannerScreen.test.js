import React from 'react';
import ScannerScreen  from './ScannerScreen';
import {
    configure,
    mount,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import  {findByTestAttr, mockPlatform, findByTestProps, mockAlert} from '../_test_/componenets/test_function';
// import Alert from 'Alert';

import { Provider } from 'react-redux';
// import {act} from 'react-dom/test-utils';
// JestHook.mock('expo-font');
import configureStore from 'redux-mock-store';
import TestRenderer from 'react-test-renderer';
const {act} = TestRenderer;
const mockStore = configureStore([]);

configure({
    adapter: new EnzymeAdapter
})
const function_click=jest.fn()

const dummy_data = {
    
    navigation:{setParams:function_click, navigate:function_click,toggleDrawer:function_click}

}


describe('<ScannerScreen />', () => {
     let store;
 
    beforeEach(() => {
      store = mockStore({
   
      });
      store.dispatch=jest.fn();
        mockAlert()  
    });
 
    afterEach(() => {
        jest.clearAllMocks();
      });
  it("snapshot test in ScannerScreen", () =>
        {
            const wrapper = mount(<Provider store={store}><ScannerScreen  {...dummy_data} /></Provider> )
            expect(wrapper).toMatchSnapshot()      
 
        });
     
        });


