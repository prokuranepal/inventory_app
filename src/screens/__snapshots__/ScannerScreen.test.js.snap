// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScannerScreen /> snapshot test in ScannerScreen 1`] = `ReactWrapper {}`;
